client/ayon_substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -89,11 +102,90 @@ def create_image_instance(self, instance, template, outputs,
         map_identifier = strip_template(template)
         suffix += f".{map_identifier}"
 
+        # Prepare representation
+        representation = {
+            "name": ext.lstrip("."),
+            "ext": ext.lstrip("."),
+            "files": fnames if len(fnames) > 1 else fnames[0],
+        }
+
+        # Mark as UDIM explicitly if it has UDIM tiles.
+        if bool(outputs[0].get("udim")):
+            # The representation for a UDIM sequence should have a `udim` key
+            # that is a list of all udim tiles (str) like: ["1001", "1002"]
+            # strings. See CollectTextures plug-in and Integrators.
+            representation["udim"] = [output["udim"] for output in outputs]
+
+        self.prepare_image_instance(
+            instance, task_entity, representation,
+            suffix, first_filepath, outputs[0],
+            texture_set_name, stack_name
+        )
+
+    def create_image_instance_by_tilename(self, instance, template, output,
+                                          task_entity, texture_set_name,
+                                          stack_name, strip_texture_set=False):
+        """Create a new instance per image by $uvTileNAme.
+
+        The new instances will be of product type `image`.
+
+        """
+        first_filepath = output["filepath"]
+        fnames = os.path.basename(output["filepath"])
+        ext = os.path.splitext(first_filepath)[1]
+        assert ext.lstrip("."), f"No extension: {ext}"
+
+        # Define the suffix we want to give this particular texture
+        # set and set up a remapped product naming for it.
+        uv_tile_name = output["uvTileName"]
+        suffix = f".{uv_tile_name}"
+        if not strip_texture_set:
+            texture_set = substance_painter.textureset.TextureSet.from_name(
+                texture_set_name
+            )
+            # More than one texture set, include texture set name
+            suffix += f".{texture_set_name}"
+            if texture_set.is_layered_material() and stack_name:
+                # More than one stack, include stack name
+                suffix += f".{stack_name}"
+
+        # Always include the map identifier
+        map_identifier = strip_template(template)
+        suffix += f".{map_identifier}"
+
+        # Prepare representation
+        representation = {
+            "name": ext.lstrip("."),
+            "ext": ext.lstrip("."),
+            "files": fnames,
+        }
+
+        # Mark as UDIM explicitly if it has UDIM tiles.
+        if bool(output.get("udim")):
+            representation["udim"] = [output["udim"]]
```

Comment:
This full logic seems shared with the other method - can't we deduplicate this? It doesn't seem unique at all.